src/corppa/poetry_detection/chadwyck_healey/tml_parser.py

```diff
@@ -0,0 +1,970 @@
+
+    # Field names for output metadata
+    metadata_fields = [
+        "filename",
+        "author_lastname",
+        "author_firstname",
+        "author_birth",
+        "author_death",
+        "author_period",
+        "transl_lastname",
+        "transl_firstname",
+        "transl_birth",
+        "transl_death",
+        "title_id",
+        "title_main",
+        "title_sub",
+        "edition_id",
+        "edition_text",
+        "period",
+        "genre",
+        "rhymes",
+    ]
+
+    def __init__(
+        self,
+        input_dir: Path,
+        output_dir: Path,
+        output_csv: Path,
+        show_progress: bool = True,
+        verbose: bool = False,
+    ):
+        self.input_dir = input_dir
+        self.output_dir = output_dir
+        self.metadata_file = output_csv
+        self.show_progress = show_progress
+        self.verbose = verbose
+
+        # Create output directory if it doesn't already exist
+        self.output_dir.mkdir(parents=True, exist_ok=True)
+
+        # keep track of files that contain only a figure
+        self.figure_only = []
+
+        # For messaging
+        self.current_file: Optional[Path] = None
+
+    def try_read_file(self, file_path: Path) -> tuple[str, str]:
+        """
+        Attempt to determine a file's text encoding and then read it
+        Parameters:
+            file_path (Path): The path to the file.
+        Returns:
+            tuple: (content of file as string, encoding used)
+        """
+        encoding = determine_encoding(file_path)
+        text = file_path.read_text(encoding=encoding)
+        return text, encoding
+
+    def clean_text(self, text: str) -> str:
+        """
+        Clean and normalize text content.
+        Steps:
+            - Collapse multiple whitespace characters into a single space.
+            - Fix common text encoding issues using ftfy.
+            - Replace known entities.
+            - Strip leading/trailing whitespace.
+        Parameters:
+            text (str): The raw text to be cleaned.
+        Returns:
+            str: Cleaned and normalized text.
+        """
+        if not text:
+            return ""
+        text = re.sub(r"\s+", " ", text)
+        text = ftfy.fix_text(text, unescape_html=True, normalization="NFKC")
+        text = replace_entities(text)
+        text = text.strip()
+        return text
+
+    def get_text_from_element(self, element: Tag) -> str:
+        """
+        Extract text from an element while preserving inline formatting.
+        This includes:
+         - Converting italic spans to plain text.
+         - Converting small caps (and similar) to uppercase.
+         - Cleaning the final output text.
+        Parameters:
+            element (Tag): A BeautifulSoup element.
+        Returns:
+            str: The cleaned text content.
+        """
+        if not element:
+            return ""
+
+        # convert italic spans (e.g., <span class="italic">word</span> => word)
```

Comment:
It seems like it would be cleaner and easier to use beautifulsoup strings instead of text, but not sure we want to change that now.